dask_jobqueue/core.py

```diff
@@ -232,10 +239,13 @@ def scale_up(self, n, **kwargs):
 
     def scale_down(self, workers):
         ''' Close the workers with the given addresses '''
-        if isinstance(workers, dict):
-            names = {v['name'] for v in workers.values()}
-            job_ids = {name.split('-')[-2] for name in names}
-            self.stop_workers(job_ids)
+        if not isinstance(workers, dict):
+            raise ValueError(
+                'Expected dictionary of workers, got %s' % type(workers))
+        names = {v['name'] for v in workers.values()}
+        # This will close down the full group of workers
+        job_ids = {name.split('-')[-2] for name in names}
+        self.stop_workers(job_ids)
```

Comment:
I'm thinking there is a better way to do this. The current behavior to scale down removes the entire job from the system. So if `Adaptive` tells us to remove 1 worker (say we have 10 workers per job), we're going to remove all 10.

@mrocklin - Would it make sense to add logic to `Adaptive` so it knows how to bundle groups of workers? Otherwise, we could bundle here and check to see we're being asked to scale down an entire group.

Reply:
The key= parameter to `workers_to_close` (passed through from `retire_workers`) seems relevant here.  I believe that it was made for this purpose.

https://github.com/dask/distributed/blob/master/distributed/scheduler.py#L2525-L2548

Reply:
Glad to see that grouped worker is handled in adaptive!

Another comment here, not linked to this PR, is that I find the `job_ids` var name misleading. Should be something like `worker_ids`.